app/src/main/java/org/kiwix/kiwixmobile/nav/destination/library/LocalLibraryFragment.kt

```diff
@@ -87,7 +88,9 @@ class LocalLibraryFragment : ZimFileSelectFragment() {
     val toolbar = root.findViewById<Toolbar>(R.id.toolbar)
     val activity = activity as AppCompatActivity
     activity.setSupportActionBar(toolbar)
+    activity.supportActionBar!!.setDisplayHomeAsUpEnabled(true)
     activity.supportActionBar!!.setTitle(R.string.library)
+    (activity as CoreMainActivity).setupDrawerToggle(toolbar)
```

Comment:
We are doing a lot of casting now, I would say make CoreMainActivity implement an interface and `onAttach` of the fragments cast the context to this interface and use this interface

Reply:
I think the casts are redundant, should only have to cast the activity variable to CoreMainActivity and nothing else.

Reply:
Yeah but this is tightly coupled in lots of places now, if you have the interface `NavDrawerActivity` then that is all you need to disentangle yourself from Core, should you need to